src/main/java/com/dataartisans/flink/dataflow/translation/wrappers/streaming/io/UnboundedSourceWrapper.java

```diff
@@ -87,6 +86,7 @@ public void run(SourceContext<WindowedValue<T>> ctx) throws Exception {
 				context.collectWithTimestamp(makeWindowedValue(item, timestamp), timestamp.getMillis());
 			}
 
+			// TODO: This will run the loop only until the underlying reader first indicates input has stalled.
 			// try to go to the next record
```

Comment:
I think what he meant in the TODO is that reader.advance() might return `false` at some point. This does not mean, however, that it will not return `true` at some point in the future.

Our implementation stops the first time the reader indicates that it currently has no data. We should never stop, since the UnboundedSource is, well, unbounded and will never stop. :smile: